commands/validators.js

```diff
@@ -0,0 +1,114 @@
+const exitOnError = require('../utils/exit-on-error');
+const connect = require('../utils/connect');
+const { utils } = require('near-api-js');
+const BN = require('bn.js');
+const AsciiTable = require('ascii-table');
+
+function findSeatPrice(validators, numSeats) {
+    let stakes = validators.map((v) => new BN(v.stake, 10)).sort((a, b) => a.cmp(b));
+    let num = new BN(numSeats);
+    let stakesSum = stakes.reduce((a, b) => a.add(b));
+    // assert stakesSum >= numSeats
+    let left = new BN(1), right = stakesSum.add(new BN(1));
+    while (!left.eq(right.sub(new BN(1)))) {
+        const mid = left.add(right).div(new BN(2));
+        let found = false;
+        let currentSum = new BN(0);
+        for (let i = 0; i < stakes.length; ++i) {
+            currentSum = currentSum.add(stakes[i].div(mid));
+            if (currentSum.gte(num)) {
+                left = mid;
+                found = true;
+                break;
+            }
+        }
+        if (!found) {
+            right = mid;
+        }
+    }
+    return left;
+}
+
+function diffValidators(currentValidators, nextValidators) {
+    const result = { newValidators: [], removedValidators: [], changeValidators: [] };
+    const validatorsMap = new Map();
+    const nextValidatorsMap = new Map();
+    nextValidators.map((validator) => nextValidatorsMap[validator.account_id] = validator);
+    currentValidators.map((validator) => { 
+        if (!(validator.account_id in nextValidatorsMap)) {
+            result.removedValidators.push(validator);
+        }
+        validatorsMap[validator.account_id] = validator;
+    });
+    nextValidators.map((validator) => {
+        if (validator.account_id in validatorsMap) {
+            if (validatorsMap[validator.account_id].stake != validator.stake) {
+                result.changeValidators.push({new: validator, old: validatorsMap[validator.account_id]});
+            }
+        } else {
+            result.newValidators.push(validator);
+        }
+    });
```

Comment:
seems like whole thing could be refactored like this:

```js
const nextValidatorsSet = new Set(nextValidators.map(v => v.account_id));
const removedValidators = currentValidators.filter(v => nextValidatorsSet.has(v.account_id));
const newValidators = nextValidators.filter(v => !validatorsSet.has(v.account_id));
// ... etc
```

in current code it's kinda hard to follow data flow and what is filtered where. E.g. is it intentional that in `nextValidators.map((validator) => {` loop some validators go neither to `changeValidators` nor to `newValidators`?

Reply:
haha, the etc is the hard part :)

Reply:
@ilblackdragon given that this change looks super useful – maybe makes sense to do some minimum cleanup first and release new near-shell:
1) add comment explaining what is happening / link to relevant near-api-js work
2) don't use `.map` as `.forEach` (i.e. just replace with `.forEach` or `for (const item of collection)` loops)
3) don't use CAPS in command